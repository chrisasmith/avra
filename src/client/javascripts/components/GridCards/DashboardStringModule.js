import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import domToImage from 'dom-to-image'
import DashboardShellModule from './DashboardShellModule'
import styles from './styles/DashboardModule.scss'

const propTypes = {
  title: PropTypes.string.isRequired,
  isFetching: PropTypes.bool.isRequired,
  value: PropTypes.string,
}

class DashboardStringModule extends Component {
  constructor(props, context) {
    super(props, context)

    this.graph = React.createRef()
    this.onDownloadImage = this.onDownloadImage.bind(this)
  }

  onDownloadImage() {
    domToImage.toJpeg(this.graph.current, { quality: 0.95 })
      .then((dataUrl) => {
        const link = document.createElement('a')
        link.download = 'nielsen-graph.jpeg'
        link.href = dataUrl
        link.click()
      })
  }

  onExpandClick(e) {
    if (e) e.preventDefault()

    this.setState({ expanded: !this.state.expanded })
  }

  render() {
    const {
      title,
      isFetching,
      value,
    } = this.props


    if (!value || isFetching) {
      return (
        <DashboardShellModule isFetching title={title} />
      )
    }

    return (
      <DashboardShellModule
        isFetching={isFetching}
        title={title}
        onDownloadCSV={this.onDownloadCSV}
        onDownloadImage={this.onDownloadImage}
      >
        <h1 ref={this.graph} className={classnames('mt-3 mb-3 text-center', styles.stringValue)}>
          {value}
        </h1>
      </DashboardShellModule>
    )
  }
}

DashboardStringModule.propTypes = propTypes
export default DashboardStringModule
