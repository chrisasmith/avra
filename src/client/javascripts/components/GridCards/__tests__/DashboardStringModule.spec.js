import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import DashboardStringModule from '../DashboardStringModule'

describe('DashboardStringModule Component', () => {
  it('Should show', () => {
    const wrapper = shallow(<DashboardStringModule
      title="title"
      isFetching={false}
      value="hi"
    />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('Should show loading', () => {
    const wrapper = shallow(<DashboardStringModule
      title="title"
      isFetching
      value="hi"
    />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
