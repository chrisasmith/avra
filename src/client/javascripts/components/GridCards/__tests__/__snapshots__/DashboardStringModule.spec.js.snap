// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardStringModule Component Should show 1`] = `
<DashboardShellModule
  isFetching={false}
  onDownloadImage={[Function]}
  title="title"
>
  <h1
    className="mt-3 mb-3 text-center stringValue"
  >
    hi
  </h1>
</DashboardShellModule>
`;

exports[`DashboardStringModule Component Should show loading 1`] = `
<DashboardShellModule
  isFetching={true}
  title="title"
/>
`;
